django/db/models/query.py

```diff
@@ -1147,7 +1147,9 @@ def in_bulk(self, id_list=None, *, field_name="pk"):
             if not id_list:
                 return {}
             filter_key = "{}__in".format(field_name)
-            batch_size = connections[self.db].features.max_query_params
+            max_params = connections[self.db].features.max_query_params or 0
+            num_fields = len(opts.pk_fields) if field_name == "pk" else 1
+            batch_size = max_params // num_fields
```

Comment:
I experimented with just:
```py
batch_size = connections[self.db].ops.bulk_batch_size([opts.pk], id_list)
```
but it caused:
```
test_bulk_batch_size (backends.sqlite.test_operations.SQLiteOperationsTests.test_bulk_batch_size) failed:

    AssertionError('1 != 500')
```
which seems overly protective. We could investigate removing that 500 limit in another PR.

Reply:
I'm good either way -- happy to go for the [simplification](https://github.com/django/django/pull/19502#discussion_r2118612666) and just live with driving more traffic into the overly protective limit until we address ticket-36430.

Reply:
I have had a think and I am happy with this approach for this backport/release blocker
I think when ticket-36430 is addressed, this could be refactored with a commit `Refs #36430, Refs #36416 --`